core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/openapi.kt

```diff
@@ -0,0 +1,2173 @@
+import org.jetbrains.kotlinx.dataframe.AnyFrame
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.DataRow
+import org.jetbrains.kotlinx.dataframe.annotations.DataSchema
+import org.jetbrains.kotlinx.dataframe.api.ConvertSchemaDsl
+import org.jetbrains.kotlinx.dataframe.api.DataSchemaEnum
+import org.jetbrains.kotlinx.dataframe.api.JsonPath
+import org.jetbrains.kotlinx.dataframe.api.KeyValueProperty
+import org.jetbrains.kotlinx.dataframe.api.asColumnGroup
+import org.jetbrains.kotlinx.dataframe.api.convert
+import org.jetbrains.kotlinx.dataframe.api.convertTo
+import org.jetbrains.kotlinx.dataframe.api.filter
+import org.jetbrains.kotlinx.dataframe.api.gather
+import org.jetbrains.kotlinx.dataframe.api.into
+import org.jetbrains.kotlinx.dataframe.api.replace
+import org.jetbrains.kotlinx.dataframe.api.with
+import org.jetbrains.kotlinx.dataframe.codeGen.AbstractDefaultReadMethod
+import org.jetbrains.kotlinx.dataframe.codeGen.CodeWithConverter
+import org.jetbrains.kotlinx.dataframe.codeGen.DefaultReadDfMethod
+import org.jetbrains.kotlinx.dataframe.codeGen.FieldType
+import org.jetbrains.kotlinx.dataframe.codeGen.GeneratedField
+import org.jetbrains.kotlinx.dataframe.codeGen.Marker
+import org.jetbrains.kotlinx.dataframe.codeGen.MarkerVisibility
+import org.jetbrains.kotlinx.dataframe.codeGen.ValidFieldName
+import org.jetbrains.kotlinx.dataframe.codeGen.isNullable
+import org.jetbrains.kotlinx.dataframe.codeGen.name
+import org.jetbrains.kotlinx.dataframe.codeGen.plus
+import org.jetbrains.kotlinx.dataframe.codeGen.toNotNullable
+import org.jetbrains.kotlinx.dataframe.codeGen.toNullable
+import org.jetbrains.kotlinx.dataframe.impl.DELIMITERS_REGEX
+import org.jetbrains.kotlinx.dataframe.impl.toCamelCaseByDelimiters
+import org.jetbrains.kotlinx.dataframe.io.AdditionalProperty.Companion.convertToAdditionalProperties
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.Any.getType
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.AnyObject.getType
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.Array.getTypeAsFrame
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.Array.getTypeAsFrameList
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.Array.getTypeAsList
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.Boolean.getType
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.Integer.getType
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.Number.getType
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.Object.getType
+import org.jetbrains.kotlinx.dataframe.io.OpenApiType.String.getType
+import org.jetbrains.kotlinx.dataframe.schema.ColumnSchema
+import java.io.File
+import java.io.InputStream
+import java.net.URL
+import kotlin.String
+import kotlin.reflect.KType
+import kotlin.reflect.full.hasAnnotation
+import kotlin.reflect.full.isSubtypeOf
+import kotlin.reflect.jvm.jvmErasure
+import kotlin.reflect.typeOf
+
```

Comment:
wdyt about splitting this file? also, at some point we started to put formats in different modules

Reply:
Okay splitting off openapi works up 'till the part where I decide to read a file as json or as openapi. Need to think of something clever or universal to make that work elegantly :)